### PR TITLE
Another attempt at fixing memory leak

### PR DESCRIPTION
**Pull request by or-else**
The previous change fixed the cases when the topic was deleted. It did not fix the most common use case of unsubscribing from all topics.

### CODE REVIEW

File: server/cluster.go [hunk 1]
```diff
@@ -744,6 +746,8 @@ func (sess *Session) rpcWriteLoop() {
 		sess.unsubAll()
 	}()
 
+	heartBeat := time.NewTimer(clusterSessionCleanup)
```
Comment (aforge) [inline server/cluster.go]: Can you add a short comment here mentioning that this is needed for GCing orphaned sessions?

**Comment by or-else** [comment]
Actually, this needs more work. The current fix has a potential for a data race:
1. Session has no subscriptions but GC is not yet triggered.
2. A new request comes for this session, while session is still in SessionStore, so `sess != nil`:
https://github.com/tinode/chat/blob/master/server/cluster.go#L355
3. Session is GC-ed before `sess.dispatch(msg.CliMsg)` is called:
https://github.com/tinode/chat/blob/master/server/cluster.go#L376
4. Message `msg.CliMsg` sent to `sess.dispatch` is lost.

**Comment by or-else** [comment]
One way to solve it is to store list of nodes in the origin session. We store list of remote nodes as a list of subscribed remote topics, so store all nodes where the origin has a proxy session.

**Comment by or-else** [comment]
Per my private comments, let's merge it with understanding that we need to refactor it. I've added comments to problematic code.

**Pull request merged**